### PR TITLE
Feature/unit test

### PR DESCRIPTION
**Pull request by nazari02105**
لطفا تست های اضافه شده مربوط به فاز ۳ را بررسی کنید
با تشکر

### CODE REVIEW

File: Phase1/src/test/java/InvertedIndexTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class InvertedIndexTest {
+    private static InvertedIndex invertedIndex;
+
+    @BeforeAll
+    public static void init() {
+        invertedIndex = new InvertedIndex();
+        invertedIndex.tokenizeFiles(new FileReader().readingFiles("src/test/testDocs"));
```
Comment (matindaghyani) [inline Phase1/src/test/java/InvertedIndexTest.java:19]: Its better to mock the FileReader object to only test the Inverted Index class, not the others.

File: Phase1/src/test/java/InvertedIndexTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Test
+    public void testSimpleInvertedIndex() {
+        TreeSet<String> queryResult = invertedIndex.query(new UserInput("Cause +People -pretty"));
```
Comment (matindaghyani) [inline Phase1/src/test/java/InvertedIndexTest.java:24]: Its better to mock the UserInput object to only test the InvertedIndex class, not the others(UserInput).

File: Phase1/src/test/java/SearchEngineTest.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+
+class SearchEngineTest {
+    @Test
+    void testSearchEngine() {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream("hello\nthis\nexit".getBytes());
+        System.setIn(inputStream);
+        new SearchEngine().run();
```
Comment (matindaghyani) [inline Phase1/src/test/java/SearchEngineTest.java]: Nothing is tested in this class! There is not assertions here.

File: Phase1/src/test/java/TestClass.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        InvertedIndex invertedIndex = new InvertedIndex().tokenizeFiles(fileReader.readingFiles("src/test/DocsForTest"));
+
+        UserInput userInput = new UserInput(" ");
+        TreeSet<String> files = invertedIndex.query(userInput);
+        Assertions.assertTrue(files.isEmpty());
+    }
+
+
+
+}
```
Comment (matindaghyani) [inline Phase1/src/test/java/TestClass.java]: This class is redundant actually. You should move methods to their relevant classes.
For example, the first method( gettingFileAndReadIt ()) should be moved to FileReader class.

**Review by matindaghyani — CHANGES_REQUESTED** [review]
Please take a look at the comments that we proposed ;-)

**Review by soroushtaslimi — CHANGES_REQUESTED** [review]
please view the points me and Matin stated and make changes accordingly. Thanks!

**Review by matindaghyani — APPROVED** [review]

**Review by soroushtaslimi — APPROVED** [review]

**Review by mhgheisarieh — APPROVED** [review]

**Review by alrz1999 — APPROVED** [review]

**Pull request merged**